contracts/swap/src/queries.rs

```diff
@@ -89,67 +128,100 @@ pub fn estimate_single_swap_execution(
         .multiplier;
     let fee_percent = market.taker_fee_rate
         * fee_multiplier
-        * (FPDecimal::one() - effective_fee_discount_rate(&market, is_self_relayer));
+        * (FPDecimal::one() - get_effective_fee_discount_rate(&market, is_self_relayer));
 
-    deps.api.debug(&format!(
-        "market.taker_fee_rate: {}, multiplier: {}, final fee percent: {}",
-        market.taker_fee_rate, fee_multiplier, fee_percent,
-    ));
+    let is_estimating_from_target = matches!(
+        swap_estimation_amount,
+        SwapEstimationAmount::ReceiveQuantity(_)
+    );
 
-    let is_buy = if balance_in.denom == market.quote_denom {
-        true
-    } else if balance_in.denom == market.base_denom {
-        false
+    let is_buy = if is_estimating_from_target {
+        balance_in.denom == market.base_denom
     } else {
-        return Err(StdError::generic_err(
-            "Invalid swap denom - neither base nor quote",
-        ));
+        balance_in.denom != market.base_denom
     };
-    deps.api.debug(&format!("Is buy: {is_buy}"));
 
-    let execution_estimate = if is_buy {
+    if is_buy {
         estimate_execution_buy(
             deps,
+            &querier,
             &env.contract.address,
             &market,
-            balance_in,
+            swap_estimation_amount,
             fee_percent,
             is_simulation,
-        )?
+        )
     } else {
-        estimate_execution_sell(deps, &querier, &market, balance_in, fee_percent)?
-    };
-    Ok(execution_estimate)
+        estimate_execution_sell(deps, &querier, &market, swap_estimation_amount, fee_percent)
+    }
 }
 
 fn estimate_execution_buy(
     deps: &Deps<InjectiveQueryWrapper>,
+    querier: &InjectiveQuerier,
```

Comment:
does it make sense to pass querier instead of initialising a new one using `deps`?